packages/gatsby/src/redux/persist.ts

```diff
@@ -26,8 +33,13 @@ function reduxChunkedNodesFilePrefix(dir: string): string {
 function reduxChunkedPagesFilePrefix(dir: string): string {
   return path.join(dir, `redux.page.state_`)
 }
+function reduxWorkerSlicesPrefix(dir: string): string {
+  return path.join(dir, `redux.worker.slices_`)
+}
 
-export function readFromCache(): ICachedReduxState {
+export function readFromCache(
+  slices?: Array<GatsbyStateSlices>
+): DeepPartial<ICachedReduxState> {
```

Comment:
Since the results for workers (using the slices) won't be the complete state I had to use `DeepPartial` in some places of this file